src/common/exception/gRPC_Result_Handling.jl

```diff
@@ -3,9 +3,9 @@ function grpc_result_or_error(result::T,
     f::Function) where {T<:Union{<:Proto.ProtoType,<:Channel{<:Proto.ProtoType},<:Nothing}}
 
     istaskdone(status) && !(fetch(status).success) &&
-        throw(TypeDBClientException(fetch(status).message, gRPCServiceCallException(fetch(status).message)))
+        throw(TypeDBClientException(fetch(status).message, gRPCServiceCallException(fetch(status).grpc_status,fetch(status).message)))
```

Comment:
This is calling `fetch(status)` 3 times. I would rewrite it with an if-statement:
```julia
if istaskdone(status)
    s = fetch(status)
    if !s.success
        throw(TypeDBClientException(s.message, gRPCServiceCallException(s.grpc_status, s.message)))
    end
end
```

Is there any reason why we want nested exception? I can understand that when `gRPCServiceCallException` is thrown by some code and then we might want to wrap it. But if this code is creating both exception objects then I wonder why we don't just make one.